pkg/apis/monitoring/v1/types.go

```diff
@@ -1039,6 +1043,44 @@ type ProberSpec struct {
 	ProxyURL string `json:"proxyUrl,omitempty"`
 }
 
+// OAuth2 allows an endpoint to authenticate with OAuth2
```

Comment:
```suggestion
// OAuth2 allows an endpoint to authenticate with OAuth2.
```
The doc generator will concatenate multiline comments together, so terminating this sentence will make more sense in the doc. Otherwise we end up with:
```
OAuth2 allows an endpoint to authenticate with OAuth2 More info: https://prometheus.io/docs/prometheus/latest/configuration/configuration/#oauth2
```